optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -0,0 +1,608 @@
+            if option_name in kwargs:
+                del kwargs[option_name]
+
+        self.lgbm_params = args[0]
+        self.train_set = args[1]
+        self.train_subset = None  # Use for sampling.
+        self.lgbm_kwargs = kwargs
+
+        # Keep original kwargs.
+        self.original_lgbm_kwargs = kwargs.copy()
+        self.original_lgbm_params = self.lgbm_params.copy()
+
+    def _suggest_early_stopping_rounds(self):
+        # type: () -> int
+
+        num_boost_round = self.lgbm_kwargs.get('num_boost_round', 1000)
+        early_stopping_rounds = min(int(num_boost_round * 0.05), 50)
+        return early_stopping_rounds
+
+    def run(self):
+        # type: () -> lgb.Booster
+        """Perform the hyperparameter-tuning with given parameters.
+
+        Returns:
+
+            booster : Booster
+                The trained Booster model.
+        """
+        # Surpress log messages.
+        if self.auto_options['verbosity'] == 0:
+            optuna.logging.disable_default_handler()
+            self.lgbm_params['verbose'] = -1
+            self.lgbm_params['seed'] = 111
+            self.lgbm_kwargs['verbose_eval'] = False
+
+        # Handling aliases.
+        _handling_alias_parameters(self.lgbm_params)
+
+        # Sampling.
+        self.sampling_train_set()
+
+        # Tuning.
+        time_budget = self.auto_options['time_budget']
+
+        self.start_time = time.time()
+        with _timer() as t:
+            self.tune_feature_fraction()
+            if time_budget is not None and time_budget > t.elapsed_secs():
+                self.best_params.update(self.get_params())
+                return self.best_booster
+
+            self.tune_num_leaves()
+            if time_budget is not None and time_budget > t.elapsed_secs():
+                self.best_params.update(self.get_params())
+                return self.best_booster
+
+            self.tune_bagging()
+            if time_budget is not None and time_budget > t.elapsed_secs():
+                self.best_params.update(self.get_params())
+                return self.best_booster
+
+            self.tune_feature_fraction_stage2()
+            if time_budget is not None and time_budget > t.elapsed_secs():
+                self.best_params.update(self.get_params())
+                return self.best_booster
+
+            self.tune_regularization_factors()
+            if time_budget is not None and time_budget > t.elapsed_secs():
+                self.best_params.update(self.get_params())
+                return self.best_booster
+
+            self.tune_min_data_in_leaf()
+            if time_budget is not None and time_budget > t.elapsed_secs():
+                self.best_params.update(self.get_params())
+                return self.best_booster
+
+            if self.enable_adjusting_lr:
+                self.tune_learning_rate()
+
+        self.best_params.update(self.get_params())
+        return self.best_booster
+
+    def sampling_train_set(self):
```

Comment:
`sample_train_set` might be more consistent with other methods' names.

Reply:
Fixed: ec045b0